source/prerequisites/postgres.txt

```diff
@@ -1,13 +1,147 @@
    :local:
    :backlinks: none
    :depth: 1
    :class: singlecol
 
-Postgres.
\ No newline at end of file
+To run sync jobs from an PostGreSQL source database, the database may 
+require some configuration changes. If Relational Migrator determines the 
+database needs configuration changes, it automatically generates a 
+SQL script with the required changes. It is recommended to have a 
+Database Administrator (DBA) review the commands in this script and 
+perform their execution on the database server. These instructions 
+configuration PostGreSQL for both types of sync jobs:
+
+.. include:: /includes/fact-short-sync-job-desc.rst
+
+About this Task
+---------------
+
+Relational Migrator uses the PostgreSQL ``pgoutput`` plugin for logical 
+decoding. This plug-in is included in PostgreSQL versions 10+, no 
+additional installation or configuration is required.
+
+Before you Begin
+----------------
+
+If PostgreSQL is configured as a cluster, Relational Migrator must 
+connect to the master server.
+
+Steps
+-----
+
+.. tabs::
+
+   .. tab:: Snapshot Jobs
+      :tabid: enable-snapshot-jobs
+
+      For snapshot jobs against PostGreSQL, the service account requires
+      schema ``USAGE`` and table ``SELECT`` permissions. Replace 
+      ``POSTGRES`` with the name of your service account user:
+
+      .. code-block:: sql
+         :copyable: true
+
+         GRANT USAGE ON SCHEMA SCHEMA_NAME TO POSTGRES;
+         GRANT SELECT ON TABLE SCHEMA_NAME.TABLE_NAME TO POSTGRES;
+
+   .. tab:: Continuous Jobs
+      :tabid: enable-continuous-jobs
+
+      .. procedure::
+         :style: normal
+
+         For continuous jobs against PostGreSQL, you must edit the 
+         configuration file, grant role permissions to the service 
+         account, and create a publication.
+
+         .. step:: Edit the configuration file 
+
+            Logical replication may not be enabled by default. To enable logical 
+            replication, change the `wal_level <https://postgresqlco.nf/doc/en/param/wal_level>`__ 
+            configuration in the 
+            `postgresql.conf configuration file <https://www.postgresql.org/docs/current/config-setting.html>`__. 
+            You *must restart the database instance* after changing the configuration file. 
+
+            .. code-block:: none
+
+               wal_level = logical 
+
+         .. step:: Create a SQL replication role
+
+            a. Create a role with ``REPLICATION`` and ``LOGIN`` 
+               database permissions:
+
+               .. code-block:: sql
+
+                  CREATE ROLE REPLICATION_GROUP REPLICATION LOGIN;
+
+            #. Grant the table ``SELECT`` and schema ``USAGE``
+               permissions to the ``REPLICATION_GROUP`` role. Each table in
+               the migration requires a ``GRANT SELECT`` statement:
+
+               .. code-block:: sql
+                  :copyable: true
+
+                  GRANT USAGE ON SCHEMA SCHEMA_NAME TO REPLICATION_GROUP;
```

Comment:
As above, can we use `<schema_name>` etc so you can tell what's a keyword and what needs to be replaced.

Reply:
Updated throughout